vm/src/builtins/genericalias.rs

```diff
@@ -0,0 +1,166 @@
+use crate::builtins::{PyStr, PyTuple, PyTupleRef, PyTypeRef};
+use crate::common::hash;
+use crate::slots::{Hashable, SlotConstructor};
+use crate::{
+    IdProtocol, PyClassImpl, PyContext, PyObjectRef, PyRef, PyResult, PyValue, TryFromObject,
+    TypeProtocol, VirtualMachine,
+};
+use std::fmt;
+
+#[pyclass(module = "types", name = "GenericAlias")]
+pub struct PyGenericAlias {
+    origin: PyTypeRef,
+    args: PyTupleRef,
+    parameters: PyTupleRef,
+}
+
+pub type PyGenericAliasRef = PyRef<PyGenericAlias>;
```

Comment:
I don't think we will have many usage of this dedicated ref type. How about just using `PyRef<PyGenericAlias>`?

Reply:
I don't have an opinion about it but I just tried to create a type like the other type does. I will remove the type `PyGenericAliasRef` as you suggested. 👌🏻